thread.c

```diff
@@ -551,18 +551,18 @@ rb_threadptr_join_list_wakeup(rb_thread_t *thread)
     while (join_list) {
         rb_thread_t *target_thread = join_list->thread;
 
-        if (target_thread->scheduler != Qnil) {
-            rb_scheduler_unblock(target_thread->scheduler, target_thread->self, rb_fiberptr_self(join_list->fiber));
+        if (target_thread->scheduler != Qnil && rb_fiberptr_blocking(join_list->fiber) == 0) {
+            rb_fiber_scheduler_unblock(target_thread->scheduler, target_thread->self, rb_fiberptr_self(join_list->fiber));
         } else {
-            rb_threadptr_interrupt(target_thread);
-
-            switch (target_thread->status) {
-                case THREAD_STOPPED:
-                case THREAD_STOPPED_FOREVER:
-                    target_thread->status = THREAD_RUNNABLE;
-                default:
-                    break;
-            }
+          rb_threadptr_interrupt(target_thread);
+
+          switch (target_thread->status) {
+              case THREAD_STOPPED:
+              case THREAD_STOPPED_FOREVER:
+                  target_thread->status = THREAD_RUNNABLE;
+              default:
+                  break;
+          }
```

Comment:
Why are these indents slipped?

Reply:
Sorry, it looks like error, I'll fix it. Thanks for catching it.